### DISCUSSION TITLE
Error: localStorage is not defined on page

### DISCUSSION THREAD

**Discussion by suyesnet** [vercel/next.js#69271] [Help]
### Summary

after uploading on on vercel, it showing my localStorage errors 

**ReferenceError: localStorage is not defined**

`"use client"
import Link from "next/link";
import { DEMO_EVENTS_LISTINGS } from "@/data/listings";
import { EventsDataType } from "@/data/types";
import EventsCard from "@/components/EventsCard";
import React, { FC, useEffect, useState } from "react";

export interface RecentlyViewedProps {
  className?: string;
  data?: EventsDataType[];
}

const RecentlyViewed: React.FC<RecentlyViewedProps> = () => {
  const [recentlyViewedEvents, setRecentlyViewedEvents] = useState<EventsDataType[]>([]);

  useEffect(() => {
    const viewedEvents = JSON.parse(localStorage.getItem('recentlyViewedEvents') || '[]');
    const events = DEMO_EVENTS_LISTINGS.filter(event => viewedEvents.includes(event.id));
    setRecentlyViewedEvents(events.slice(0, 4));
  }, []);

  return (
    <div>
      <h2>Recently Viewed Events</h2>
      {recentlyViewedEvents.length > 0 ? (
        <div className="grid grid-cols-1 gap-5 md:gap-6 sm:grid-cols-2 lg:grid-cols-4 2xl:grid-cols-4">
          {recentlyViewedEvents.map((event) => (
            <EventsCard key={event.id} data={event} />
          ))}
        </div>
      ) : (
        <p>No recently viewed events found.</p>
      )}
    </div>
  );
};

export default RecentlyViewed;`

### Additional information

```js
what could be the issue here ?
```

**Comment by ZoranRavic** [comment]
That error happens when you try to use `localStorage` outside `useEffect`.

This is not the case in the example you provided, so you'll have to look elsewhere in your code. The error should have included the file name and line number.
